### PR TITLE
Use containerd in DIND scenario

### PR DESCRIPTION
**Pull request by nmiekley**
use containerd instead of docker in [DIND scenario](https://madhuakula.com/kubernetes-goat/docs/scenarios/scenario-2/docker-in-docker-exploitation-in-kubernetes-containers)

# Motivation 
Moved from [Discussion](https://github.com/madhuakula/kubernetes-goat/discussions/127) to Pull Request:

The original idea of the scenario is that docker is running on the worker nodes and managing the containers for kubernetes. However, with [kubernetes deprecation for docker shim](https://kubernetes.io/blog/2020/12/02/dockershim-faq/) this is no longer the case. Now, either containerd or cri-o is used as container runtime for kubernetes.

The same exploit can still be done, we only have to use the containerd. I tested it on AWS EKS and kind, but of course for a cluster with cri-o a different socket needs to be mounted. 

I am not sure how to name the updated scenario so I am open to suggestion. 

Btw, the current setup also has a funny effect in KIND: here the docker daemon you gain control over is the one on your machine running the kind node. You can still get control of the KIND node and the containerd inside, but that is not really a production relevant scenario. This pull request also avoids the special treatment of kind and you now correctly gain control of the daemon in the cluster nodes. 

This would probably solve this: https://github.com/madhuakula/kubernetes-goat/discussions/127

### CODE REVIEW

File: guide/docs/scenarios/images/sc-2-6.png
Comment (st4rtr3ck) [inline guide/docs/scenarios/images/sc-2-6.png:1]: I believe the new image could have a slightly higher pixel count to match the old one better.

File: guide/docs/scenarios/images/sc-4-3.png
Comment (st4rtr3ck) [inline guide/docs/scenarios/images/sc-4-3.png:1]: I fear the new screen shot could be overwhelming with too much information in too little space on small screens.

File: guide/docs/scenarios/scenario-2/scenario-2.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ By the end of the scenario, we will understand and learn the following
 
 ### ⚡️ The story
 
-Most of the CI/CD and pipeline systems use the underlying host Docker runtime to build containers for you within the pipeline by using something called DIND (docker-in-docker) with a UNIX socket. Here in this scenario, we try to exploit this misconfiguration and gain access to the host system by escaping out of the docker container.
+Most of the CI/CD and pipeline systems use the underlying host container runtime to build containers for you within the pipeline by using something called DIND (docker-in-docker) with a UNIX socket. Here in this scenario, we try to exploit this misconfiguration and gain access to the host system of the worker node by escaping out of the docker container.
```
Comment (st4rtr3ck) [inline guide/docs/scenarios/scenario-2/scenario-2.md:25]: Thanks for the changes here @nmiekley

File: guide/docs/scenarios/scenario-2/scenario-2.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ Most of the CI/CD and pipeline systems use the underlying host Docker runtime to
 
 ### 🎯 Goal
 
-The goal of this scenario is to escape out of the running docker container to the host system where the container is running and able to access and perform actions on the host system.
+The goal of this scenario is to escape out of the running docker container to the host system where the container is running and able to access and perform actions on other container running on the same node.
```
Comment (st4rtr3ck) [inline guide/docs/scenarios/scenario-2/scenario-2.md:37]: nitpick: better pluralize: "other container**s**"

File: guide/docs/scenarios/scenario-2/scenario-2.md [hunk 3]
```diff
@@ -89,52 +72,53 @@ www.google.com.
...
 
 ![Scenario 2 mount](../images/sc-2-3.png)
 
-- Wow! we can see the `/custom/docker/docker.sock` mounted in the file system and assuming it's mounted from the host system we need to talk to it for communicating with the UNIX socket
+- Wow! we can see the `/custom/containerd/containerd.sock` mounted in the file system and assuming it's mounted from the host system we need to talk to it for communicating with the UNIX socket
 
 :::tip
 
-We can use multiple methods for communicating with the `docker.sock` UNIX socket. Some of them include [official docker binary](https://download.docker.com/linux/static/stable/), or a simple `curl` program as well.  
+We can use multiple methods for communicating with the `containerd.sock` UNIX socket. Some of them include [crictl binary](https://github.com/kubernetes-sigs/cri-tools/blob/master/docs/crictl.md), or a simple `curl` program as well.
```
Comment (st4rtr3ck) [inline guide/docs/scenarios/scenario-2/scenario-2.md:87]: nitpick: don't we want an article here, like "include a|the crictl binary"

**Review by st4rtr3ck — APPROVED** [review]
Nice work. Thanks for the update.

**Comment by madhuakula** [comment]
Thanks for the amazing work @nmiekley @st4rtr3ck 🙏

**Pull request merged**